src/huggingface_hub/inference/_providers/cohere.py

```diff
@@ -0,0 +1,17 @@
+from huggingface_hub.inference._providers._common import (
+    BaseConversationalTask,
+)
+
+
+_PROVIDER = "cohere"
+_BASE_URL = "https://api.cohere.com"
+
+
+class CohereConversationalTask(BaseConversationalTask):
+    def __init__(self):
+        super().__init__(provider=_PROVIDER, base_url=_BASE_URL)
+
+    def _prepare_route(self, mapped_model: str) -> str:
+        if self.task == "conversational":
+            return "/compatibility/v1/chat/completions"
+        raise ValueError(f"Unsupported task '{self.task}' for Cohere API.")
```

Comment:
A user is not supposed to override `self.task` here, and we already have a check if the task is supported or not here: [huggingface_hub/inference/_providers/_common.py#L119](https://github.com/huggingface/huggingface_hub/blob/87c3eaf686c0d12234db8f3ee5f7a6488bfe5981/src/huggingface_hub/inference/_providers/_common.py#L119).
So something like this should be good enough:
```suggestion
    def _prepare_route(self, mapped_model: str) -> str:
            return "/compatibility/v1/chat/completions"
```